### DISCUSSION TITLE
Getting the following error when hosted in a VM

### DISCUSSION THREAD

**Discussion by damithcl** [ludeknovy/jtl-reporter#208] [Q&A] [answered]
Getting the following error for the http://{IP}:2020/api/info call  
500
`{message: "Unexpected error occurred: d8f6add6-0b9d-443f-9b2b-fabdd8557d53"}
message
: 
"Unexpected error occurred: d8f6add6-0b9d-443f-9b2b-fabdd8557d53"`

**Comment by ludeknovy** [comment]
Hi @damithcl was this happening rigth after you docker-composed' it up, or put differently, did it happen after you launched it?

**Reply by damithcl** [reply]
It happens while trying to launch it . docker-compose is fine .  VM -  Ubuntu 18.04 .  jtl-reporter version 4.8.1 . back-end error is as follows 
`jtl-reporter-db | 2023-03-29 03:34:48.429 UTC [126] ERROR:  relation "jtl.users" does not exist at character 15
jtl-reporter-db | 2023-03-29 03:34:48.429 UTC [126] STATEMENT:  SELECT * FROM jtl.users
be_1         | {"code":"42P01","file":"parse_relation.c","length":108,"level":"error","line":"1352","name":"error","position":"15","routine":"parserOpenTable","severity":"ERROR"}
be_1         | {"level":"error","message":"Unexpected error: error: relation \"jtl.users\" does not exist, errorId: 321afaab-4f4c-4956-aed4-ff0df79b5d95"}
fe_1         | 10.212.134.190 - - [29/Mar/2023:03:34:48 +0000] "GET /api/info HTTP/1.1" 500 77 "http://{ip}:2020/login?returnUrl=%2Fdashboard" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36"
be_1         | {"level":"info","message":"GET /api/info 500 108ms","meta":{}}`

**Reply by ludeknovy** [reply]
Hi, it looks like the migration was completed successfully. Run `docker-compose up` and leave it running for a while. Also make sure you havent modified the docker-compose file, and the migration service is present.

**Reply by damithcl** [reply]
Please find the entire log here .Still the issue exist without modifying the docker-compose file.
`Creating network "jtl-reporter_default" with the default driver
Creating jtl-reporter_be_1        ... done
Creating jtl-reporter_listener_1  ... done
Creating jtl-reporter_scheduler_1 ... done
Creating jtl-reporter-db          ... done
Creating jtl-reporter_fe_1        ... done
Creating jtl-reporter_migration_1 ... done
Attaching to jtl-reporter-db, jtl-reporter_be_1, jtl-reporter_listener_1, jtl-reporter_scheduler_1, jtl-reporter_fe_1, jtl-reporter_migration_1
be_1         | 
be_1         | > jtl-reporter-be@1.0.0 start
be_1         | > node ./dist/server.js
be_1         | 
jtl-reporter-db | 
jtl-reporter-db | PostgreSQL Database directory appears to contain a database; Skipping initialization
jtl-reporter-db | 
be_1         | {"level":"info","message":"Express server listening on port 5000"}
be_1         | {"level":"info","message":"Bree scheduler was started"}
listener_1   | 
listener_1   | > jtl-reporter-async-app@1.0.0 start
listener_1   | > node ./dist/server.js
listener_1   | 
be_1         | {"level":"info","message":"By using this app you agree with the use of analytics in this app to help improve user experience and the overall functionality of the app."}
fe_1         | /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
fe_1         | /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
jtl-reporter-db | 2023-03-29 07:30:14.105 UTC [1] LOG:  starting PostgreSQL 13.4 on x86_64-pc-linux-musl, compiled by gcc (Alpine 10.3.1_git20210424) 10.3.1 20210424, 64-bit
fe_1         | /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
fe_1         | 10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
scheduler_1  | 
scheduler_1  | > jtl-reporter-scheduler@1.0.0 start
scheduler_1  | > node dist/index.js
scheduler_1  | 
listener_1   | {"level":30,"time":1680075018097,"pid":17,"hostname":"a8cd184ec601","msg":"Server listening at http://0.0.0.0:6000"}
jtl-reporter-db | 2023-03-29 07:30:14.106 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
fe_1         | 10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
fe_1         | /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
fe_1         | /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
jtl-reporter-db | 2023-03-29 07:30:14.108 UTC [1] LOG:  listening on IPv6 address "::", port 5432
jtl-reporter-db | 2023-03-29 07:30:14.113 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
fe_1         | /docker-entrypoint.sh: Configuration complete; ready for start up
jtl-reporter-db | 2023-03-29 07:30:14.136 UTC [21] LOG:  database system was shut down at 2023-03-29 07:29:57 UTC
jtl-reporter-db | 2023-03-29 07:30:14.199 UTC [1] LOG:  database system is ready to accept connections
jtl-reporter-db | 2023-03-29 07:30:14.241 UTC [27] LOG:  TimescaleDB background worker launcher connected to shared catalogs
migration_1  | 
migration_1  | > jtl-reporter-be@1.0.0 migrate
migration_1  | > node-pg-migrate up
migration_1  | 
migration_1  | > Migrating files:
migration_1  | > - 1638448762711_delete-samples
migration_1  | > - 1638448762740_scenario-zero-error
migration_1  | > - 1638992168018_project-stats-settings
migration_1  | > - 1641637266222_test-housekeeping
migration_1  | > - 1643273224321_role
migration_1  | > - 1643281231047_share-token-user
migration_1  | > - 1643705218437_failure-message
migration_1  | > - 1644908662609_test-run-name
migration_1  | > - 1644924773281_scenario-generate-share-token
migration_1  | > - 1646391467756_scenario-label-filter-settings
migration_1  | > - 1646733443241_label-trend-chart-settings
migration_1  | > - 1647428257976_extra-plot-data
migration_1  | > - 1668498913817_upsert-scenario
migration_1  | > - 1668684730568_user-project-permission
migration_1  | > - 1668706422783_user-project-permission-data
migration_1  | > - 1669967522715_user-request-stats
migration_1  | > - 1673372364043_apdex
migration_1  | > - 1673597363690_apdex-item
migration_1  | > - 1674942200154_histogram-plot
migration_1  | > - 1677248456262_user-scenario-settings-trends
migration_1  | > - 1677248456270_samples-timestamp
migration_1  | > - 1677248456271_scatter-plot-data-column
migration_1  | ### MIGRATION 1638448762711_delete-samples (UP) ###
migration_1  | ALTER TABLE "jtl"."scenario"
migration_1  |   ADD "delete_samples" boolean DEFAULT false NOT NULL;
migration_1  | INSERT INTO "public"."pgmigrations" (name, run_on) VALUES ('1638448762711_delete-samples', NOW());
migration_1  | 
migration_1  | 
jtl-reporter-db | 2023-03-29 07:30:26.429 UTC [40] ERROR:  schema "jtl" does not exist
jtl-reporter-db | 2023-03-29 07:30:26.429 UTC [40] STATEMENT:  ALTER TABLE "jtl"."scenario"
jtl-reporter-db | 	  ADD "delete_samples" boolean DEFAULT false NOT NULL;
migration_1  | Error executing:
migration_1  | ALTER TABLE "jtl"."scenario"
migration_1  |   ADD "delete_samples" boolean DEFAULT false NOT NULL;
migration_1  | error: schema "jtl" does not exist
migration_1  | 
migration_1  | > Rolling back attempted migration ...
migration_1  | error: schema "jtl" does not exist
migration_1  |     at Parser.parseErrorMessage (/src/be/node_modules/pg-protocol/dist/parser.js:287:98)
migration_1  |     at Parser.handlePacket (/src/be/node_modules/pg-protocol/dist/parser.js:126:29)
migration_1  |     at Parser.parse (/src/be/node_modules/pg-protocol/dist/parser.js:39:38)
migration_1  |     at Socket.<anonymous> (/src/be/node_modules/pg-protocol/dist/index.js:11:42)
migration_1  |     at Socket.emit (node:events:513:28)
migration_1  |     at addChunk (node:internal/streams/readable:324:12)
migration_1  |     at readableAddChunk (node:internal/streams/readable:297:9)
migration_1  |     at Readable.push (node:internal/streams/readable:234:10)
migration_1  |     at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
migration_1  |   length: 93,
migration_1  |   severity: 'ERROR',
migration_1  |   code: '3F000',
migration_1  |   detail: undefined,
migration_1  |   hint: undefined,
migration_1  |   position: undefined,
migration_1  |   internalPosition: undefined,
migration_1  |   internalQuery: undefined,
migration_1  |   where: undefined,
migration_1  |   schema: undefined,
migration_1  |   table: undefined,
migration_1  |   column: undefined,
migration_1  |   dataType: undefined,
migration_1  |   constraint: undefined,
migration_1  |   file: 'namespace.c',
migration_1  |   line: '3050',
migration_1  |   routine: 'get_namespace_oid'
migration_1  | }
migration_1  | npm notice 
migration_1  | npm notice New major version of npm available! 8.19.3 -> 9.6.2
migration_1  | npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.6.2>
migration_1  | npm notice Run `npm install -g npm@9.6.2` to update!
migration_1  | npm notice 
jtl-reporter_migration_1 exited with code 1
`

**Reply by ludeknovy** [reply]
Thanks for the logs!
` error: schema "jtl" does not exist migration_1 ` this means the DB was not properly initialized for some reason – that happens automatically on the first run. My advice here is to start again one more time. Delete the checkout folder / checkout the repository into another folder and run docker-compose up -d again.

**Reply by ludeknovy** [reply]
@damithcl Did you make it work?

**Reply by damithcl** [reply]
Not yet . it seems the SQL  file  has a permission issue in  the container . When executed as a fresh copy it gives the SQL file permission denied error.

**Reply by ludeknovy** [reply]
@damithcl ah, ok, so this is not the application problem itself then. Have a look at docker volume permission issue - maybe here? https://mydeveloperplanet.com/2022/10/19/docker-files-and-volumes-permission-denied/

**Reply by damithcl** [reply, marked as answer]
Hi @ludeknovy  , I have tried with adding the permission to Dockerfile but was not successful.Permission changes anre not persisting in the docker volume .  For example I have tried like ,
`FROM timescale/timescaledb:2.4.1-pg13
USER root
ENV POSTGRES_DB jtl_report
COPY --chown=user:user schema.sql /docker-entrypoint-initdb.d/
RUN chmod -R 765 /docker-entrypoint-initdb.d/`

Finally I got it done via manually executing the sql within the container.
`psql  -U postgres -d jtl_report -a -f /docker-entrypoint-initdb.d/schema.sql `

There may be a issue with Ubuntu 18.04 docker permissions but I m not sure . It all works fine in 20.04.

Thanks a lot for your tremendous support on this :)

**Reply by ludeknovy** [reply]
Hi @damithcl Glad you solved it after all!
If you happen to have any questions/ feedback, feel free to use the discussion section again.
